components/runes/src/db/cache/rune_validation.rs

```diff
@@ -0,0 +1,554 @@
+use std::str::FromStr;
+
+use bitcoin::{Script, Transaction, Witness};
+use bitcoind::{
+    bitcoincore_rpc,
+    utils::{
+        bitcoind::{bitcoin_get_raw_transaction, bitcoind_get_block_height},
+        Context,
+    },
+};
+use config::BitcoindConfig;
+use ordinals_parser::{Rune, Runestone};
+
+fn unversioned_leaf_script_from_witness(witness: &Witness) -> Option<&Script> {
+    #[allow(deprecated)]
+    witness.tapscript()
+}
+
+/// Reserved runes cannot be etched because they serve as a fallback mechanism for failed etchings
+/// and prevent namespace collision in the Bitcoin Runes protocol. When an etching transaction fails
+/// to specify a valid rune name or the etching is malformed, the protocol automatically assigns a
+/// "reserved" rune name using a deterministic formula to maintain protocol integrity.
+pub fn is_reserved(rune: &Rune) -> bool {
+    rune.0 >= Rune::RESERVED
+}
+
+pub async fn tx_commits_to_rune(
```

Comment:
Change the name to something more descriptive e.g. `rune_etching_has_valid_commit`. Also add a documentation comment to this function.